x/pubkey/client/cli/tx.go

```diff
@@ -0,0 +1,187 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	cmtjson "github.com/cometbft/cometbft/libs/json"
+	cmtos "github.com/cometbft/cometbft/libs/os"
+
+	"cosmossdk.io/core/address"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	crypto "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/server"
+
+	"github.com/sedaprotocol/seda-chain/x/pubkey/types"
+)
+
+const (
+	SEDAKeyFileName = "seda_keys.json"
+
+	// FlagKeyFile defines a flag to specify an existing key file.
+	FlagKeyFile = "key-file"
+)
+
+// GetTxCmd returns the CLI transaction commands for this module
+func GetTxCmd(valAddrCodec address.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      fmt.Sprintf("%s transactions subcommands", types.ModuleName),
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+	cmd.AddCommand(
+		AddKey(valAddrCodec),
+	)
+	return cmd
+}
+
+// AddKey returns the command for adding a new key and uploading its
+// public key on chain at a given index.
+func AddKey(ac address.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add-key",
```

Comment:
Maybe we should update the name and descriptions to indicate it generates all keys and submits a TX to register the new public keys?